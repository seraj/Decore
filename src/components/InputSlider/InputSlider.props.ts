import { InputRangeProps } from "react-input-range";
import ThemeProps from "../../theme/Theme.props";

export default interface InputSliderProps
  extends ThemeProps,
    React.HTMLAttributes<HTMLDivElement> {}

export interface InputRangeProps extends ThemeProps, InputRangeProps {}
