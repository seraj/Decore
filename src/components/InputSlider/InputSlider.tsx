import * as React from "react";
import InputRange from "react-input-range";
import StyledInputSlider from "./styled/StyledInputSlider";
import InputSliderProps, { InputRangeProps } from "./InputSlider.props";
import "react-input-range/lib/css/index.css";

const Container: React.FC<InputSliderProps> = ({ children, ...props }) => (
  <StyledInputSlider {...props}>{children}</StyledInputSlider>
);
const InputSlider: React.FC<InputRangeProps> = ({ ...props }) => (
  <Container>
    <InputRange {...props} />
  </Container>
);

export default InputSlider;
