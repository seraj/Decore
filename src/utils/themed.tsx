import * as React from "react";
import * as renderer from "react-test-renderer";
import Decore from "../components/Decore";
import theme from "../theme";

const themed = (component: React.ReactChild) =>
  renderer.create(<Decore theme={theme}>{component}</Decore>);

export default themed;
