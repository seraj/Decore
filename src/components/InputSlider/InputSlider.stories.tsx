import { number, withKnobs, boolean } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { storiesOf } from "@storybook/react";
import * as React from "react";
import wInfo from "../../utils/wInfo";
import InputSlider from "./InputSlider";

const stories = storiesOf("Components/InputSlider", module);
stories.addDecorator(withKnobs);

stories.add(
  "default",
  wInfo()(() => (
    <span style={{ padding: 40, display: "block" }}>
      <InputSlider
        maxValue={number("maxValue", 20)}
        minValue={number("minValue", 0)}
        disabled={boolean("disabled", false)}
        value={number("value", 4)}
        step={number("step", 2)}
        onChange={action("onChange")}
      />
    </span>
  ))
);
