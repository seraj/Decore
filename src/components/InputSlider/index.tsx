import InputSlider from "./InputSlider";

export default InputSlider;
