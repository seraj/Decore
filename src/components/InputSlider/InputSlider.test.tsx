import * as React from "react";

import themed from "../../utils/themed";
import InputSlider from "./InputSlider";

describe("InputSlider", () => {
  it.skip("should render correctly", () => {
    const component = themed(
      <InputSlider
        onChange={() => null}
        value={1}
        minValue={1}
        maxValue={2}
        disabled={false}
        step={1}
      />
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
