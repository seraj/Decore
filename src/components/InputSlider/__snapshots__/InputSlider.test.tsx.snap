// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSlider should render correctly 1`] = `
.c0 .input-range__slider,
.c0 .input-range__track--active {
  background: #333333;
  border-color: #333333;
}

<div
  className="c0"
/>
`;
